test/BasicEditingTest.VI.cs

```diff
@@ -1121,5 +1121,87 @@ public void ViInsertModeMoveCursor()
                 _.RightArrow, // 'RightArrow' again does nothing, but doesn't crash
                 "c"));
         }
+
+        [SkippableFact]
+        public void ViDefect1281_1()
+        {
+            TestSetup(KeyMode.Vi);
+
+            Test("bcd", Keys(
+                "abcdabcd", _.Escape,
+
+                // return to the [B]eginning of the word,
+                // then [c]hange text un[t]il just before the [2]nd [b] character
+                // this leaves the cursor at the current position (0) but erases
+                // the "abcda" text portion, / switches to edit mode and
+                // positions the cursor just before the "bcd" text portion.
+
+                "Bc2tb",
+
+                // going back to normal mode again without having modified the buffer further
+                // even though the [c] command started an edit group, going back to normal
+                // mode closes the pending edit group.
+
+                _.Escape, CheckThat(() => AssertCursorLeftIs(0)),
+
+                // attempt to [c]hange text un[t]il just before the [2]nd [b] character again
```

Comment:
I would slightly change the wording of the comments which no longer reflect the exact behaviour following this PR. Something like:

```patch
  // attempt to [c]hange text un[t]il just before the [2]nd [b] character again
  // because the [b] character only appears once further down in the buffer
  // relative to where the cursor position is – currently set to 0 - the command
  // fails. Therefore, we are still in normal mode.
  //
-  // however, even though the command failed, it still started an edit group
-  // which is now pending further edit actions
+ // as the command failed, the current edit group is now correctly closed.

  "c2tb", CheckThat(() => AssertLineIs("bcd")),

  // attempt to [c]hange text un[t]il just before the [2]nd [b] character a third time.
```